core/shared/src/main/scala/zio/Chunk.scala

```diff
@@ -753,7 +753,19 @@ object Chunk {
    * Returns a chunk backed by an array.
    */
   def fromArray[A](array: Array[A]): Chunk[A] =
-    if (array.isEmpty) Empty else Arr(array)
+    (if (array.isEmpty) Empty
+     else
+       array.asInstanceOf[AnyRef] match {
+         case x: Array[AnyRef]  => AnyRefArray(x)
```

Comment:
Have you tested to make sure Scala is doing the right thing here w.r.t type erasure? Ordinarily you can't match on, e.g., `List[Int]` because of type erasure. There is a means to do this for arrays, because they store their type inside them, but I'm not 100% sure if Scala utilizes this to make this type casing sane. Worth testing to be sure.

Or to avoid you could match on `Array#elemTag` which gives you the tag of the element type.

Reply:
I’ve tested in the REPL that a byte array becomes a ByteArray

Reply:
The pattern matching works here because an Array[Int] is a distinct type at runtime, whereas List[Int] is just a List[AnyRef] at runtime

Reply:
Ok, so Scalac has baked this in. 👍